src/net_permissions.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2009-2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <net_permissions.h>
+#include <util/system.h>
+#include <netbase.h>
+
+bool TryParsePermissionFlags(const std::string str, NetPermissionFlags& output, size_t& readen, std::string* error)
```

Comment:
Use `std::string&` (see recommendation in developer notes).

Nit: Use another variable name to avoid shadowing `error` included from `util/system.h`.

Reply:
I was using pointer because I don't want to burden the user to get the error. But meh... everywhere we use it we need the error so I will change to ref.